app/models/test_passage.rb

```diff
@@ -62,8 +57,6 @@ def correct_answers
   end
 
   def next_question
-    return nil if current_question.nil?
-
     test.questions.order(:id).where('id > ?', current_question.id).first
```

Comment:
why you remove this condition?
return nil if current_question.nil?

the comment was for this line
test.questions.order(:id).where('id > ?', current_question.id).first

))))